src/iosrv/init.c

```diff
@@ -256,7 +256,75 @@ dss_topo_init()
 }
 
 static int
-server_init()
+abt_max_num_xstreams(void)
+{
+	char   *env;
+
+	env = getenv("ABT_MAX_NUM_XSTREAMS");
+	if (env == NULL)
+		env = getenv("ABT_ENV_MAX_NUM_XSTREAMS");
+	if (env != NULL)
+		return atoi(env);
+	return 0;
+}
+
+static int
+set_abt_max_num_xstreams(int n)
+{
+	char   *name = "ABT_MAX_NUM_XSTREAMS";
+	char   *value;
+	int	rc;
+
+	D_ASSERTF(n > 0, "%d\n", n);
+	rc = asprintf(&value, "%d", n);
```

Comment:
There's a D_ASPRINTF() macro for this, it allows tracing of malloc/free across the application so as this function stands it would throw an error as you're logging a free, but not a allocation.

Reply:
There's a series of patches changing "asprintf, ..., free" to "asprintf, ..., D_FREE". I don't like the asymmetry either. I've adopted D_ASPRINTF here.